### PR TITLE
test: add TransformUtils edge case tests for blank name and unusual ISO3 codes

### PR DESCRIPTION
**Pull request by replicant1**
- [x] Add `regionToRegionCodePreservesUnusualCharacters` — verifies non-alpha characters in ISO3 code pass through unchanged
- [x] Add `regionToRegionCodePreservesEmptyString` — verifies empty ISO3 code passes through unchanged
- [x] Add `regionEntityToRegionPreservesBlankName` — verifies blank name passes through unchanged
- [x] Rename all three new test methods from snake_case to camelCase to match existing naming convention
- [x] Update `test_gaps.txt` to mark all three gaps as resolved

### CODE REVIEW

File: app/src/test/java/com/rodbailey/covid/domain/TransformUtilsTest.kt [hunk 1]
```diff
@@ -97,4 +97,28 @@ class TransformUtilsTest {
             regionAus.toRegionCode().chars
         )
     }
+
+    @Test
+    fun regionEntityToRegion_preservesBlankName() {
+        assertEquals(
+            Region(iso3Code = "AUS", name = ""),
+            RegionEntity(iso3code = "AUS", name = "").toRegion()
```
[the remark this answers is not in the document]
Comment (replicant1) [reply app/src/test/java/com/rodbailey/covid/domain/TransformUtilsTest.kt:105]: @copilot apply changes based on [this feedback](https://github.com/replicant1/Covid/pull/34#discussion_r3019535814)
